cognite/client/data_classes/hosted_extractors/sources.py

```diff
@@ -0,0 +1,636 @@
+        if isinstance(self.authentication, MQTTAuthenticationWrite):
+            output["authentication"] = self.authentication.dump(camel_case)
+        if isinstance(self.ca_certificate, CACertificateWrite):
+            output["caCertificate" if camel_case else "ca_certificate"] = self.ca_certificate.dump(camel_case)
+        if isinstance(self.auth_certificate, AuthCertificateWrite):
+            output["authCertificate" if camel_case else "auth_certificate"] = self.auth_certificate.dump(camel_case)
+        return output
+
+
+@dataclass
+class MQTTAuthentication(CogniteObject, ABC):
+    _type: ClassVar[str]
+
+    @classmethod
+    @abstractmethod
+    def _load_authentication(cls, resource: dict[str, Any]) -> Self:
+        raise NotImplementedError()
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        type_ = resource.get("type")
+        if type_ is None and hasattr(cls, "_type"):
+            type_ = cls._type
+        elif type_ is None:
+            raise KeyError("type")
+
+        authentication_class = _MQTTAUTHENTICATION_CLASS_BY_TYPE.get(type_)
+        if authentication_class is None:
+            return UnknownCogniteObject(resource)  # type: ignore[return-value]
+        return cast(Self, authentication_class._load_authentication(resource))
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case)
+        output["type"] = self._type
+        return output
+
+
+@dataclass
+class BasicMQTTAuthentication(MQTTAuthentication):
+    _type = "basic"
+    username: str
+
+    @classmethod
+    def _load_authentication(cls, resource: dict[str, Any]) -> Self:
+        return cls(username=resource["username"])
+
+
+@dataclass
+class CACertificate(CogniteObject):
+    thumbprint: str
+    expires_at: str
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(thumbprint=resource["thumbprint"], expires_at=resource["expiresAt"])
+
+
+@dataclass
+class AuthCertificate(CogniteObject):
+    thumbprint: str
+    expires_at: str
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(thumbprint=resource["thumbprint"], expires_at=resource["expiresAt"])
+
+
+class _MQTTSource(Source, ABC):
+    def __init__(
+        self,
+        external_id: str,
+        host: str,
+        created_time: int,
+        last_updated_time: int,
+        port: int | None = None,
+        authentication: MQTTAuthentication | None = None,
+        useTls: bool = False,
+        ca_certificate: CACertificate | None = None,
+        auth_certificate: AuthCertificate | None = None,
+    ) -> None:
+        super().__init__(external_id)
+        self.host = host
+        self.port = port
+        self.authentication = authentication
+        self.useTls = useTls
```

Comment:
There are several more occurrences
```suggestion
        use_tls: bool = False,
        ca_certificate: CACertificate | None = None,
        auth_certificate: AuthCertificate | None = None,
    ) -> None:
        super().__init__(external_id)
        self.host = host
        self.port = port
        self.authentication = authentication
        self.use_tls = use_tls
```